Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -267,14 +267,39 @@ extension BrowserViewController: WKNavigationDelegate {
       
       let domainForMainFrame = Domain.getOrCreate(forUrl: mainDocumentURL, persistent: !isPrivateBrowsing)
       
-      // Debouncing logic
+      // Debouncing and query param stripping
       // Handle debouncing for main frame only and only if the site (etld+1) changes
       // We also only handle `http` and `https` requests
       if Preferences.Shields.autoRedirectTrackingURLs.value, url.isWebPage(includeDataURIs: false),
          let currentURL = tab?.webView?.url,
          currentURL.baseDomain != url.baseDomain,
          domainForMainFrame.isShieldExpected(.AdblockAndTp, considerAllShieldsOption: true),
          navigationAction.targetFrame?.isMainFrame == true {
+        // Handle query param stripping
+        if let requestURL = navigationAction.request.url, let requestMethod = navigationAction.request.httpMethod {
+          let filteredURL = QueryFilterService.stripQueryParams(
```

Comment:
Instead of returning the existing URL that was passed in, this should return `Optional<URL>` instead since the Brave-Core API can return `null`. Then you can just check if the result is null, and if it is, do nothing. If it isn't, redirect.

Reply:
Sure that's a good idea.